experimental/traffic-portal/src/app/core/origins/table/origins-table.component.ts

```diff
@@ -0,0 +1,147 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { Component, type OnInit } from "@angular/core";
+import { FormControl } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute } from "@angular/router";
+import { BehaviorSubject } from "rxjs";
+import { TypeFromResponse } from "trafficops-types";
+import { ResponseOrigin } from "trafficops-types/dist/origin";
+
+import { OriginService } from "src/app/api";
+import { CurrentUserService } from "src/app/shared/current-user/current-user.service";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import type {
+	ContextMenuActionEvent,
+	ContextMenuItem,
+	DoubleClickLink
+} from "src/app/shared/generic-table/generic-table.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * OriginsTableComponent is the controller for the "Origins" table.
+ */
+@Component({
+	selector: "tp-origins",
+	styleUrls: ["./origins-table.component.scss"],
+	templateUrl: "./origins-table.component.html"
+})
+export class OriginsTableComponent implements OnInit {
+	/** List of origins */
+	public origins: Promise<Array<ResponseOrigin>>;
+
+	/** Definitions of the table's columns according to the ag-grid API */
+	public columnDefs = [
+		{
+			field: "name",
+			headerName: "Name"
+		},
+		{
+			field: "tenant",
+			headerName: "Tenant"
```

Comment:
you should use a valueGetter or something to add in the ID of the Tenant. Unfortunately, that's still sometimes more important to know than the name.